src/Strategy/Conda/EnvironmentYml.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda.EnvironmentYml (
+  analyze',
+  buildGraph,
+  EnvironmentYmlFile(..)
+) where
+
+import Effect.Exec
+import Effect.ReadFS
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import Control.Carrier.Diagnostics hiding (fromMaybe)
+import Graphing (Graphing, fromList)
+import Data.Aeson
+import Data.Text (Text)
+import Data.Maybe (fromMaybe)
+import Data.List.Extra ((!?))
+import Path
+import Types
+
+
+buildGraph :: EnvironmentYmlFile -> Graphing Dependency
+buildGraph envYmlFile = Graphing.fromList (map toDependency allDeps)
+  where
+    allDeps = map getCondaDepFromText (dependencies envYmlFile)
+    toDependency :: CondaDep -> Dependency
+    toDependency CondaDep{..} =
+       Dependency 
+       {
+        dependencyType = CondaType,
+        dependencyName = depName,
+        dependencyVersion = CEq <$> depVersion, -- todo - properly handle version constraints
+        dependencyLocations = [],
+        dependencyEnvironments = [],
+        dependencyTags = M.empty
+      }
+
+analyze' :: 
+ ( Has ReadFS sig m
+ , Has Diagnostics sig m
+ )
+  => Path Abs File -> m (Graphing Dependency)
+analyze' envYml = buildGraph <$> readContentsYaml @EnvironmentYmlFile envYml
+
+
+-- an example Text: "biopython=1.78=py38haf1e3a3_0"
+-- '=' is a delimeter for <name>=<version>=<build>
+-- where <version> and <build> are optional
+getCondaDepFromText :: Text -> CondaDep
+getCondaDepFromText rcd =
+  CondaDep
+  { depName = name
+  , depVersion = version
+  -- Note these aren't currently being used
+  , depBuild = build
+  , depFullVersion = fullVersion
+  }
+  
+  where
+    depSplit = T.split (== '=') rcd
+
+    name = fromMaybe rcd (depSplit !? 0)
+    version = depSplit !? 1 -- TODO: this may contain constraints that we need to parse
+    build = depSplit !? 2
+    fullVersion = getFullVersion version build
+
+    -- if we have a version AND a build, we combine them to form a full version
+    -- i.e. Just "1.2.3" Just "build" => Just "1.2.3=build"
```

Comment:
[nit]: More haddock styling.  We don't do much of this, but I think that we should where it's easy.
[reference for doc examples](https://haskell-haddock.readthedocs.io/en/latest/markup.html#examples)
```suggestion
    >>> getFullVersion (Just "1.2.3") (Just "build")
    Just "1.2.3=build"
```